src/components/SearchBar.js

```diff
@@ -1,15 +1,51 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import { searchPhoto } from '../actions';
 
-const SearchBar = () => {
-    return (
-        <div className="ui loading search">
-            <div className="ui icon input">
-                <input className="prompt" type="text" placeholder="Search..." />
-                <i className="search icon"></i>
-            </div>
-            <div className="results"></div>
-        </div>
-    );
+class SearchBar extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = { searchTerm: null };
+    }
+
+    onSearchChange = (event) => {
+        var searchTerm = event.target.value;
+
+        this.setState({ searchTerm });
+    }
+
+    onSearchSubmit = (event) => {
+        event.preventDefault();
+
+        this.props.searchPhoto(this.state.searchTerm);
```

Comment:
We have redux connected. You should be storing these values there.

Reply:
This is the approach I'm thinking. Let me know what you think: 
1. OnChange in Input of the `SearchBar`, call a different action to change the state of `searchTerm`
2. OnSubmit take the current state of `searchTerm` then call the `SEARCH_PHOTO` action to change the state of `photos`
3. Use `this.getState()` to rerender the `ImageDetail` component

Reply:
Can you please clarify. What do you mean by `call a different action`. Different from what?

Reply:
I suggest that there is a redux action that is used to set search term in redux state. You can then define a selector to grab that info and pull it into you component state.

I agree that you can dispatch an action handler that handles a fetch call also. You may want to dispatch several actions. On that captures a fetch was sent, one that shows it was receive, and one that shows it failed.

I'm not sure why you would use getState(). if you connect the component you'll have access to the state, when it updates you can use something like `componentDidRecievePorps` to handle any additional actions you want to fire.

Reply:
I wasn't sure when `getState()` actually gets used. Still trying to figure out the standard patterns and structures for react/redux apps.

Also, I've refactored the onChange handler of the SearchBar to call a reducer to change the state of the `searchTerm`. Check this commit and let me know what you think: https://github.com/Shenanigans510/yeet/pull/2/commits/4436147150102641f09f1e990150b1db11f39dac